python/scrabble-score/scrabble_score.py

```diff
@@ -1,23 +1,19 @@
 def score(word):
 
-    word = word.lower()
-    letters = {}
-
     scores = {
-        1: ['a', 'e', 'i', 'o', 'u', 'l', 'n', 'r', 's', 't'],
-        2: ['d', 'g'],
-        3: ['b', 'c', 'm', 'p'],
-        4: ['f', 'h', 'v', 'w', 'y'],
-        5: ['k'],
-        8: ['j', 'x'],
-        10: ['q', 'z']
+        1: 'aeioulnrst',
+        2: 'dg',
+        3: 'bcmp',
+        4: 'fhvwy',
+        5: 'k',
+        8: 'jx',
+        10: 'qz'
     }
 
     total = 0
-    for l in word:
+    for l in word.lower():
         for pts, group in scores.items():
             if l in group:
-                letters[l] = pts
```

Comment:
By the construction of scoring groups, you expect this condition to be true exactly once.
Hence there is no need for the weird dictionary, we can just accumulate the score as we go.
Moreover, once the solution is found, we don't need to check the remaining groups: break out of the inner loop.
Given that the groups with low score are more popular, this could actually improve performance as well.